### PR TITLE
Subject Detail page

### PR DESCRIPTION
**Pull request by Jimmy-Tempest**
image

**Features**
- Connecting Class to a subject
- Editing a room subject
- Deleting a room subject

**Fixes**
- Class and class advisor for both student and teacher are now fetched from the database

### CODE REVIEW

File: src/components/dialogs/ConnectSubject.tsx [hunk 1]
```diff
@@ -21,6 +21,11 @@ import { nameJoiner } from "@utils/helpers/name";
 import { ChipInputListItem, SubmittableDialogProps } from "@utils/types/common";
 import { Teacher } from "@utils/types/person";
 import { Subject, SubjectListItem } from "@utils/types/subject";
+import { supabase } from "@utils/supabaseClient";
+import { ClassroomTable } from "@utils/types/database/class";
+import { getCurrentAcedemicYear } from "@utils/helpers/date";
+import { useTeacherAccount } from "@utils/hooks/auth";
+import { RoomSubjectTable } from "@utils/types/database/subject";
```
Comment (SiravitPhokeed) [inline src/components/dialogs/ConnectSubject.tsx:28]: Imports should be grouped.

File: src/pages/t/subjects/[subjectID].tsx [hunk 2]
```diff
@@ -776,8 +796,10 @@ const SubjectDetails: NextPage<{
         show={deleteConnection.show}
         onClose={() => setDeleteConnection({ show: false })}
         onSubmit={() => {
-          setDeleteConnection({ show: false });
-          router.replace(router.asPath);
+          handleDelete().then(() => {
+            setDeleteConnection({ show: false });
+            router.replace(router.asPath);
+          });
```
Comment (SiravitPhokeed) [inline src/pages/t/subjects/[subjectID].tsx:802]: It may be possible to use an async function here.

File: src/utils/backend/database.ts [hunk 3]
```diff
@@ -112,7 +125,6 @@ export async function db2Teacher(teacher: TeacherDB): Promise<Teacher> {
         th: teacher.SubjectGroup.name_th,
       },
     },
-    // TODO: Fetch contact
```
Comment (SiravitPhokeed) [inline src/utils/backend/database.ts:115]: It's so cool seeing TODOs removed. That's all!

**Review by SiravitPhokeed — APPROVED** [review]
Fetch functions in components could be moved to a separate file in the `/utils/backend/` folder for increased readability for the components. But that's for later™.

_LGTM!_

**Review by ImSadudee — APPROVED** [review]
LGTM!

**Pull request merged**